main/pipeline/pipeline.cc

```diff
@@ -406,49 +406,65 @@ void incrementStrictLevelCounter(core::StrictLevel level) {
-    prodCounterInc("types.input.files");
-    if (core::File::isRBIPath(fileName)) {
-        counterAdd("types.input.rbi.bytes", src.size());
-        counterInc("types.input.rbi.files");
-    }
+    switch (file.dataAllowingUnsafe(gs).sourceType) {
+        case core::File::Type::NotYetRead: {
+            string fileName{file.dataAllowingUnsafe(gs).path()};
+            Timer timeit(gs.tracer(), "readFileWithStrictnessOverrides", {{"file", fileName}});
+            string src;
+            bool fileFound = true;
+            try {
+                src = opts.fs->readFile(fileName);
+            } catch (FileNotFoundException e) {
+                // continue with an empty source, because the
+                // assertion below requires every input file to map
+                // to one output tree
+                fileFound = false;
+            }
+            prodCounterAdd("types.input.bytes", src.size());
+            prodCounterInc("types.input.files");
+            if (core::File::isRBIPath(fileName)) {
+                counterAdd("types.input.rbi.bytes", src.size());
+                counterInc("types.input.rbi.files");
+            }
 
-    {
-        core::UnfreezeFileTable unfreezeFiles(gs);
-        auto fileObj = make_shared<core::File>(move(fileName), move(src), core::File::Type::Normal);
-        // Returns nullptr if tree is not in cache.
-        ast = fetchTreeFromCache(gs, file, *fileObj, kvstore);
+            {
+                core::UnfreezeFileTable unfreezeFiles(gs);
+                auto fileObj = make_shared<core::File>(move(fileName), move(src), core::File::Type::Normal);
 
-        auto entered = gs.enterNewFileAt(move(fileObj), file);
-        ENFORCE(entered == file);
-    }
-    if (enable_counters) {
-        counterAdd("types.input.lines", file.data(gs).lineCount());
-    }
+                auto entered = gs.enterNewFileAt(move(fileObj), file);
+                ENFORCE(entered == file);
+            }
 
-    auto &fileData = file.data(gs);
-    if (!fileFound) {
-        if (auto e = gs.beginError(sorbet::core::Loc::none(file), core::errors::Internal::FileNotFound)) {
-            e.setHeader("File Not Found");
+            if constexpr (enable_counters) {
+                counterAdd("types.input.lines", file.data(gs).lineCount());
+            }
+
+            if (!fileFound) {
+                if (auto e = gs.beginError(sorbet::core::Loc::none(file), core::errors::Internal::FileNotFound)) {
+                    e.setHeader("File Not Found");
+                }
+            }
+
+            break;
         }
+        case core::File::Type::Normal: {
+            // If we load successfully from the cache then there weren't any indexing errors, and otherwise we'll set
+            // this flag during indexing if necessary.
+            file.data(gs).setHasIndexErrors(false);
+
+            break;
+        }
+        case core::File::Type::PayloadGeneration:
+        case core::File::Type::Payload:
```

Comment:
What should the behavior be for payload files?

Is it possible that our changes might end up with us needing to re-generate an AST for a payload file?

Reply:
I think the current behavior is what we want: we'll only have source for them if the user of an LSP session has explicitly opened them, and we will always reindex that source given that it won't get cached.

Reply:
I ran this on Stripe's codebase, and opened up some of the stdlib rbis. The behavior seemed consistent with existing behavior: there weren't any features missing once I had opened an RBI.